src/services/base/labeling.ts

```diff
@@ -67,3 +67,18 @@ export function addClassificationLabels(projectId: string, recordId: string, lab
     }
     jsonFetchWrapper(finalUrl, FetchType.POST, onResult, JSON.stringify(convertCamelToSnakeCase(body)));
 }
+
+export function addExtractionLabel(projectId: string, recordId: string, labelingTaskId: string, tokenStartIndex: number, tokenEndIndex: number, value: string, labelId: string, asGoldStar: boolean, sourceId: string, onResult: (result: any) => void) {
+    const finalUrl = `${labelingEndpoint}/${projectId}/add-extraction-label`;
+    const body = {
+        recordId,
+        labelingTaskId,
+        tokenStartIndex,
+        tokenEndIndex,
+        value,
+        labelId,
+        asGoldStar,
+        sourceId
+    }
+    jsonFetchWrapper(finalUrl, FetchType.POST, onResult, JSON.stringify(convertCamelToSnakeCase(body)));
+}
```

Comment:
<img width="726" alt="Screenshot 2024-04-15 at 15 39 26" src="https://github.com/code-kern-ai/refinery-ui/assets/153752112/79e47dab-33fc-4daa-b122-665da1553d62">

Currently not allowing me to add highlighted words, seems its stuck in a certain position.